### DISCUSSION TITLE
Getting clangd to use more resources

### DISCUSSION THREAD

**Discussion by xim** [clangd/clangd#1270] [Q&A]
I have a rather massive project that I'm indexing with clangd (>10k files) and indexing can take a while. I often index while my computer is running a large amount of work, often stuff like `nice ionice my-build-command`

While that build is running, my editor is snappy but the clangd often uses less than even a single CPU. My computer has a lot of CPU cores, and I wouldn't mind letting clang use at least half of them for indexing, as that's more important to me in many cases.

I'm running clangd 15, a build from late june, and tried setting `-j $(nproc) --background-index-priority=normal` at the command line but still the clangd background indexer often stays below 50% CPU usage while I'd love it if it could eat at least half of my available CPU...

Is there something I'm missing here?

**Comment by xim** [comment]
I've mitigated this *slightly* by adding an even higher nice value: `nice -n19 ionice my-build-command`

Would still like to see more resource usage on clangd. This just gets me slightly more CPU on clangd =)

**Reply by sergei-dyshel** [reply]
I'm having this problem too. Basically, since indexing (for C++ template-heavy files) is not more CPU-intensive than compilation, I would expect indexing the entire project taking not more time than project build (assuming build is just compilation/linking). In reality indexing entire project takes lots of time. 

Is there a way to make indexing faster and take all available CPU?
